clients/mobile/src/LocationService/LocationService.ts

```diff
@@ -40,4 +63,84 @@ export class LocationService {
       }
     }
   }
+
+  async preferedLocationAutocomplete(location: string, city: string): Promise<Array<{ label: string; value: string }>> {
+    const COUNTRY = 'BD'
+
+    try {
+      const response = await this.httpClient.get<{ predictions: Prediction[] }>('/place/autocomplete/json', {
+        input: location,
+        types: 'geocode',
+        components: `country:${COUNTRY}`,
+        key: GOOGLE_MAP_API_KEY
+      })
+      const formattedData = response.predictions
+        .filter((prediction: Prediction) => {
+          return (
+            (prediction.types.includes('neighborhood') || prediction.types.includes('political'))
+          )
+        })
+        .map((prediction: Prediction) => ({
+          label: prediction.description,
+          value: prediction.description
+        }))
+
+      return formattedData
+    } catch (error) {
+      console.error('Error fetching location autocomplete:', error)
+      return []
```

Comment:
in other places you are throwing errors; here, you're just returning empty array. curious

Reply:
Instead of returning an empty array now throw an error